### DISCUSSION TITLE
Tabs limit

### DISCUSSION THREAD

**Discussion by xfzv** [sxyazi/yazi#1646] [4. Ideas]
Currently, it's possible to create up to 9 tabs:

https://github.com/sxyazi/yazi/blob/49639aa34ca69fc72008dbae6a2ad033d888805f/yazi-core/src/manager/commands/tab_create.rs#L7

https://github.com/sxyazi/yazi/blob/49639aa34ca69fc72008dbae6a2ad033d888805f/yazi-core/src/manager/commands/tab_create.rs#L32-L34

Just curious what's the rationale behind this limitation.

Any chance tabs could be unlimited in the future? I'm aware most users won't open 9+ tabs simultaneously but it can be useful in some cases.

**Comment by AnirudhG07** [comment]
Using numbers to jump to tabs, would be only limited to 9. Although you can always use the next and previous tab key bindings. But say you opened 20 tabs, no you wont be able to jump to 15, etc directly since you (according to current set of keybindings) have max 9. Although some rules can be set for it, but the need to extend would be when someone actually needs it, and as you said, you dont need too many tabs at once, 9 is good enough

**Reply by AnirudhG07** [reply]
There are many possible features that could be possibly added, but would be implemented if someone actually needs them.

**Reply by xfzv** [reply]
For instance, Dolphin Explorer allows unlimited tabs and has `Alt+{1,9}` keybindings to go to tab 1-9. A valid concern is that tabs aren't clickable in `yazi`, so one would have to use previous/next tab keys to focus tab 15 as you say, indeed.
